proto/spire/api/agent/delegation/v1/delegation.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+package spire.api.agent.delegation.v1;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/agent/delegation/v1;delegationv1";
+
+import "spire/api/types/selector.proto";
+import "spire/api/types/x509svid.proto";
+
+// The deletation service provides an interface to get the SVIDs of other
+// workloads on the host. This service is intended for use cases where a process
+// (different than the workload one) should access the workload's SVID to
+// perform actions on behalf of the workload. One example of is using a single
+// node instance of Envoy that upgrades TCP connections for different processes
+// running in such a node.
+//
+// The caller must be local and its identity must be listed in the allowed
+// clients on the spire-agent configuration.
+service Delegation {
+    // Fetch X.509-SVIDs for multiple workloads.
+    rpc FetchX509SVIDs(stream FetchX509SVIDsRequest) returns (stream FetchX509SVIDsResponse);
+
+    // Fetch local and federated bundles.
+    rpc FetchX509Bundles(FetchX509BundlesRequest) returns (stream FetchX509BundlesResponse);
+}
+
+// X.509 SPIFFE Verifiable Identity Document with the private key.
+message X509SVIDWithKey {
+    spire.api.types.X509SVID x509_svid = 1;
+    bytes x509_svid_key = 2;
```

Comment:
Key encoding? I recommend DER PKCS#8

Reply:
Are you saying we should define this encoding (DER PKCS#8) in the comment?

Reply:
Yes, I think that is what @evan2645 intends. Otherwise it isn't immediately clear to callers how to parse the result.